pkg/kwokctl/cmd/portforward/portforward.go

```diff
@@ -0,0 +1,133 @@
+/*
+Package portforward provides commands for forwarding a local port to a port on a pod.
+It supports different runtime environments such as binary, Docker, Podman, Nerdctl, and kind.
+This is useful for accessing component ports that are not actively exposed when the cluster is created.
+*/
+package portforward
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"sigs.k8s.io/kwok/pkg/apis/internalversion"
+	"sigs.k8s.io/kwok/pkg/config"
+	"sigs.k8s.io/kwok/pkg/consts"
+	"sigs.k8s.io/kwok/pkg/kwokctl/runtime"
+	"sigs.k8s.io/kwok/pkg/log"
+	"sigs.k8s.io/kwok/pkg/utils/exec"
+	"sigs.k8s.io/kwok/pkg/utils/path"
+)
+
+type flagpole struct {
+	Name      string
+	LocalPort string
+	*internalversion.KwokctlConfiguration
+}
+
+// NewCommand returns a new cobra.Command for port-forward in a cluster
+func NewCommand(ctx context.Context) *cobra.Command {
+	flags := &flagpole{}
+
+	cmd := &cobra.Command{
+		Use:   "port-forward NAME LOCAL_PORT",
+		Short: "Forward a local port to a port on a pod",
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			flags.Name = config.DefaultCluster
+			flags.LocalPort = args[1]
+			err := runE(cmd.Context(), flags, args)
+			if err != nil {
+				return fmt.Errorf("%v: %w", args, err)
+			}
+			return nil
+		},
+	}
+	cmd.DisableFlagParsing = true
+	return cmd
+}
+
+func runE(ctx context.Context, flags *flagpole, args []string) error {
+	name := config.ClusterName(flags.Name)
+	workdir := path.Join(config.ClustersDir, flags.Name)
+
+	logger := log.FromContext(ctx)
+	logger = logger.With("cluster", flags.Name)
+	ctx = log.NewContext(ctx, logger)
+
+	rt, err := runtime.DefaultRegistry.Load(ctx, name, workdir)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			logger.Warn("Cluster does not exist")
+		}
+		return err
+	}
+
+	switch flags.Options.Runtime {
```

Comment:
Move the implementation to its own package.
https://github.com/kubernetes-sigs/kwok/tree/main/pkg/kwokctl/runtime